api/envoy/extensions/filters/network/http_connection_manager/v3/http_connection_manager.proto

```diff
@@ -870,6 +870,10 @@ message HttpConnectionManager {
   // This should be set to `false` in cases where Envoy's view of the downstream address may not correspond to the
   // actual client address, for example, if there's another proxy in front of the Envoy.
   google.protobuf.BoolValue add_proxy_protocol_connection_state = 53;
+
+  // If set to true the HCM will set the last round trip time (RTT) into connection info of the downstream connection
+  // after every request. Defaults to `false`.
+  bool set_rtt_into_connection = 57;
```

Comment:
what you mean by `code-level opt-in`? A runtime guard that default to `false`? Both ok for me, I just think most users shouldn't pay the overhead for they needn't, esp the overhead is non-trivial. Even we can remove it directly if no one need it for now.

cc @DiazAlan could you take a look when you have free time, thanks?

Reply:
I was thinking that instead of a config knob, there be a function on the hcm (or somewhere) that code could call to enable this feature: `void enableRttIntoConnection(bool enable)` or something.

Reply:
then users want this feature need call this method by rebuilding envoy or call this method in their fork?

Reply:
IMO, I think we can disable this feature by runtime flag first. And the one who wants the feature could complain if they want better way.

Reply:
This could be be added to `substitution_formatter`.

Reply:
There is a similar request for it https://github.com/envoyproxy/envoy/issues/27391, maybe we should enable the access log to access the stats directly.

Reply:
> There is a similar request for it #27391, maybe we should enable the access log to access the stats directly.

And indeed we collect the tcp_rtt at https://github.com/envoyproxy/envoy/blob/cacd32bc9e3285f6eee7efbc3fcb3e7291975055/source/extensions/transport_sockets/tcp_stats/tcp_stats.cc#L154

Reply:
I think this PR was opened to disable per-request rtt refreshment to avoid high CPU overhead. Not about how to expose the rtt to end users 🤣

Reply:
If we enable the ability for access log to get the stats, then this feature can be enabled/disabled by enabled/disabled the tcp_stats extensions. Then it avoids all the users paying for this CPU overhead even if they don't need this feature.